pkg/operator/ceph/version/version.go

```diff
@@ -293,3 +297,18 @@ func ValidateCephVersionsBetweenLocalAndExternalClusters(localVersion, externalV
 
 	return nil
 }
+
+// RegisterImageVersion stores the CephVersion detected for a specified image for global access.
+func RegisterImageVersion(image string, version CephVersion) {
+	imageToVersionMapLock.Lock()
```

Comment:
Generally speaking, golang recommends using `defer` to ensure a lock is released in case there is a panic in the middle of it. For example:
```
imageToVersionMapLock.Lock()
defer imageToVersionMapLock.Unlock()
imageToVersionMap[image] = version
```